### PR TITLE
Migrate LedgerClient test from to Canton

### PR DESCRIPTION
**Pull request by remyhaemmerle-da**

### CODE REVIEW

File: ledger/ledger-api-client/src/it/scala/com/digitalasset/ledger/client/LedgerClientIT.scala [hunk 1]
```diff
@@ -22,26 +20,19 @@ import org.scalatest.wordspec.AsyncWordSpec
...
-    with Matchers
-    with Inside
-    with AkkaBeforeAndAfterAll
-    with SuiteResourceManagementAroundEach
-    with SandboxFixture {
+final class LedgerClientIT extends AsyncWordSpec with Matchers with Inside with CantonFixture {
 
-  private val LedgerId =
-    domain.LedgerId(s"${classOf[LedgerClientIT].getSimpleName.toLowerCase}-ledger-id")
+  private val LedgerId = domain.LedgerId(config.ledgerIds(0))
```
Comment (samuel-williams-da) [inline ledger/ledger-api-client/src/it/scala/com/digitalasset/ledger/client/LedgerClientIT.scala:25]: oh, I guess a question of opinion, do we prefer `(0)` or `.head`

**Review by samuel-williams-da — APPROVED** [review]

File: ledger/ledger-api-client/src/it/scala/com/digitalasset/ledger/client/CommandClientIT.scala [hunk 2]
```diff
@@ -58,35 +55,48 @@ import scala.concurrent.duration.FiniteDuration
...
   private val defaultCommandClientConfiguration =
     CommandClientConfiguration(
       maxCommandsInFlight = 1,
       maxParallelSubmissions = 1,
       defaultDeduplicationTime = Duration.ofSeconds(30),
     )
 
-  private val testLedgerId = domain.LedgerId("ledgerId")
-  private val testNotLedgerId = domain.LedgerId("hotdog")
+  private val testLedgerId = domain.LedgerId("participant0")
```
[the remark this answers is not in the document]
Comment (remyhaemmerle-da) [reply ledger/ledger-api-client/src/it/scala/com/digitalasset/ledger/client/CommandClientIT.scala:80]: :+1:

**Review by samuel-williams-da — APPROVED** [review]
Following a review from Stephen, I think we prefer `.headOption.value` with `import org.scalatest.OptionValues._` over simply `.head` for tests. Up to you, happy to merge

**Comment by remyhaemmerle-da** [comment]
> Following a review from Stephen, I think we prefer `.headOption.value` with `import org.scalatest.OptionValues._` over simply `.head` for tests, as it gives a better error message. Up to you, happy to merge

Let's do that everywhere as a separate PR.

**Pull request merged**